cmd/snap-confine/snap-confine.c

```diff
@@ -434,19 +426,12 @@ int main(int argc, char **argv)
 							real_uid,
 							real_gid, saved_gid);
 	}
-
-	// Temporarily drop privs back to calling user (we'll permanently drop
-	// after loading seccomp).
-	if (setegid(real_gid) != 0)
-		die("setegid failed");
-	if (seteuid(real_uid) != 0)
-		die("seteuid failed");
-
-	if (real_gid != 0 && geteuid() == 0)
-		die("dropping privs did not work");
-	if (real_uid != 0 && getegid() == 0)
-		die("dropping privs did not work");
-	// Ensure that the user data path exists.
+	// Ensure that the user data path exists. When creating it use the identity
+	// of the calling user (by using real user and group identifiers). This
+	// allows the creation of directories inside ~/ on NFS with root_squash
+	// attribute.
```

Comment:
While the removed comments weren't perfect, the new ones leave out a detail worth adding back in. I suggest:
```
	// Temporarily drop privileges back to the calling user until we can
	// permanently drop (which we can't do just yet due to seccomp, see
	// below).
	sc_set_effective_identity((sc_identity) {.uid = real_uid,.gid = real_gid }
	);

	// Ensure that the user data path exists. Since we've dropped privileges
	// this allows the creation of directories inside ~/ on NFS with root_squash
	// attribute.
	setup_user_data();
...
```

Reply:
Done